### PR TITLE
Add XML comments to controllers

### PR DESCRIPTION
**Pull request by sober607**
1. Added requested roles annotations, to manipulate with endpoint
2. Added XML comments to every endpoint of all controllers
3. Added example requests and responses to every endpoint of all controllers

[example screenshot](https://ibb.co/W0D5mm9)

#164

### CODE REVIEW

File: CharlieBackend.Api/Controllers/StudentGroupsController.cs [hunk 1]
```diff
@@ -53,9 +67,16 @@ public async Task<ActionResult<StudentGroupDto>> PostStudentGroup(CreateStudentG
...
+        /// Updates given student group
+        /// </summary>
+        /// <response code="200">Successful update of student group</response>
+        /// <response code="HTTP: 422, API: 4">Error, given group name already exist</response>
+        /// <response code="HTTP: 400, API: 0">Error, given group data is wrong</response>
+        [SwaggerResponse(200, type: typeof(UpdateStudentGroupDto))]
         [Authorize(Roles = "Secretary, Mentor, Admin")]
         [HttpPut("{id}")]
-        public async Task<ActionResult<StudentGroupDto>> PutStudentGroup(long id, UpdateStudentGroupDto studentGroupDto)
+        public async Task<ActionResult<UpdateStudentGroupDto>> PutStudentGroup(long id, [FromBody]UpdateStudentGroupDto studentGroupDto)
```
Comment (SnipeFrost) [inline CharlieBackend.Api/Controllers/StudentGroupsController.cs:79]: why after update you back UpdateStudentGroupDto and not StudentGroupDto?

File: CharlieBackend.Api/Controllers/StudentGroupsController.cs [hunk 2]
```diff
@@ -71,13 +92,40 @@ public async Task<ActionResult<StudentGroupDto>> PutStudentGroup(long id, Update
...
 
+        /// <summary>
+        /// Update students of stunde group
+        /// </summary>
+        /// <response code="200">Successful update of student group students</response>
+        /// <response code="HTTP: 400, API: 0">Error, given data is wrong</response>
+        [SwaggerResponse(200, type: typeof(UpdateStudentsForStudentGroup))]
+        [Authorize(Roles = "Secretary, Mentor, Admin")]
+        [HttpPut("{id}/students")]
+        public async Task<ActionResult<UpdateStudentsForStudentGroup>> PutStudentsOfStudentGroup(long id, [FromBody]UpdateStudentsForStudentGroup studentGroupDto)
```
Comment (SnipeFrost) [inline CharlieBackend.Api/Controllers/StudentGroupsController.cs:103]: same here

File: CharlieBackend.Api/Controllers/StudentGroupsController.cs [hunk 3]
```diff
@@ -71,13 +92,40 @@ public async Task<ActionResult<StudentGroupDto>> PutStudentGroup(long id, Update
             return updatedStudentGroup.ToActionResult();
         }
 
+        /// <summary>
+        /// Update students of stunde group
+        /// </summary>
+        /// <response code="200">Successful update of student group students</response>
+        /// <response code="HTTP: 400, API: 0">Error, given data is wrong</response>
+        [SwaggerResponse(200, type: typeof(UpdateStudentsForStudentGroup))]
```
Comment (SnipeFrost) [inline CharlieBackend.Api/Controllers/StudentGroupsController.cs:100]:
```suggestion
        [SwaggerResponse(200, type: typeof(UpdateStudentForStudentGroup))]
```
 wrong name a bit

File: CharlieBackend.Api/Controllers/StudentGroupsController.cs (hunk 2 above)
Comment (SnipeFrost) [inline CharlieBackend.Api/Controllers/StudentGroupsController.cs:103]: same here with name, check pls carefully

Comment (sober607) [reply CharlieBackend.Api/Controllers/StudentGroupsController.cs]: I had old version on local side. Fixed

File: CharlieBackend.Api/Controllers/StudentGroupsController.cs (hunk 3 above)
Comment (sober607) [reply CharlieBackend.Api/Controllers/StudentGroupsController.cs]: I had old version on local side. Fixed

**Review by SnipeFrost — APPROVED** [review]
good

**Review by BohdanBBG — APPROVED** [review]

**Review by SnipeFrost — APPROVED** [review]

**Review by SerjKocherga — APPROVED** [review]

**Pull request merged**